nemo/collections/common/parts/preprocessing/collections.py

```diff
@@ -190,6 +194,12 @@ def __init__(
             else:
                 data.sort(key=lambda entity: entity.duration)
 
+        if shard_strategy == 'scatter' and world_size > 1:
```

Comment:
Why duplicate data ? It will take multiple times the memory.

Reply:
Change the default behavior is better. Such duplication of data just for length calculation is too wasteful